nucypher/blockchain/eth/interfaces.py

```diff
@@ -249,14 +251,48 @@ def connect(self, fetch_registry: bool = True, sync_now: bool = False):
         else:
             self.attach_middleware()
 
-        # Wait for chaindata sync
-        if sync_now:
-            self.client.sync()
-
         # Establish contact with NuCypher contracts
         if not self.registry:
             self._configure_registry(fetch_registry=fetch_registry)
 
+        # Wait for chaindata sync
+        if sync_now:
+            sync_state = self.client.sync()
+            if emitter:
+                import click
+                emitter.echo(f"Syncing: {self.client.chain_name.capitalize()}. Waiting for sync to begin.")
+
+                while not len(self.client.peers):
+                    emitter.echo("waiting for peers...")
+                    time.sleep(5)
+
+                peer_count = len(self.client.peers)
+                emitter.echo(f"Found {'an' if peer_count == 1 else peer_count} Ethereum peer{('s' if peer_count>1 else '')}.")
+
+                try:
+                    emitter.echo("Beginning sync...")
+                    initial_state = next(sync_state)
+                except StopIteration:  # will occur if no syncing needs to happen
+                    emitter.echo("Local blockchain data is already synced.")
+                    return True
+
+                prior_state = initial_state
+                total_blocks_to_sync = int(initial_state.get('highestBlock', 0)) - int(initial_state.get('currentBlock', 0))
+                with click.progressbar(
```

Comment:
👍 . Also, if you don't want ETA to be shown with the `progressbar` because of variability in syncing of blocks, you could also pass `show_eta=False`

Reply:
In testing yesterday it is actually turned out to be very accurate, other than speeding up a bit at the end.